src/seedu/addressbook/data/person/Name.java

```diff
@@ -60,5 +60,14 @@ public boolean equals(Object other) {
     public int hashCode() {
         return fullName.hashCode();
     }
+    
+    /**
+     * Returns true of the other name is very similar to this name.
+     * Two names are considered similar if ...
+     */
+    public boolean isSimilar(Name other) {
+        
```

Comment:
In the first step, you should make the test case first. Later, it will give you an error and just by then you make this method